terraso_backend/apps/shared_data/forms.py

```diff
@@ -55,34 +55,35 @@ class Meta:
             "created_by",
         )
 
-    def is_valid_mime_type(self, data_file):
+    def validate_file(self, data_file):
         file_extension = pathlib.Path(data_file.name).suffix
-        content_type = data_file.content_type
-        file_mime_type = magic.from_buffer(data_file.open("rb").read(2048), mime=True)
-        allowed_file_extensions = mimetypes.guess_all_extensions(file_mime_type)
+        file_mime_type = magic.from_buffer(data_file.read(2048), mime=True)
+        guessed_allowed_file_extensions = mimetypes.guess_all_extensions(file_mime_type)
 
-        is_valid = (
-            file_mime_type and allowed_file_extensions and file_extension in allowed_file_extensions
-        )
+        if file_extension not in settings.DATA_ENTRY_ACCEPTED_TYPES.keys():
+            raise ValidationError(file_extension[1:], code="invalid_not_accepted_extension")
 
-        # Also check variations of allowed csv types
-        return (
-            is_valid
-            or (
-                file_extension == ".csv"
-                and content_type in VALID_CSV_TYPES
-                and file_mime_type in VALID_CSV_TYPES
-            ),
-            f"Invalid file extension ({file_extension}) for the file type {file_mime_type}",
+        guessed_is_valid = (
```

Comment:
issue: please use a more descriptive name than `guessed_is_valid`. I think this is a check based on mime time/file extension without reading the contents. is that correct?